### ISSUE TITLE
Duplicate diagnostics with invalid overloaded TypeIs

### ISSUE THREAD

**Issue by jelle-openai** [astral-sh/ty#3968]
### Summary

On this code:

```python
from typing import Never, overload
from typing_extensions import TypeIs

@overload
def is_dataclass(
    value: Never,
) -> TypeIs[bool]: ...
@overload
def is_dataclass(
    value: object,
) -> TypeIs[str]: ...
def is_dataclass(value: object) -> bool:
    return True
```

ty emits

```
Narrowed type `bool` is not assignable to the declared parameter type `Never` (invalid-type-guard-definition) [Ln 7, Col 6]
Narrowed type `bool` is not assignable to the declared parameter type `Never` (invalid-type-guard-definition) [Ln 11, Col 6]
Narrowed type `bool` is not assignable to the declared parameter type `Never` (invalid-type-guard-definition) [Ln 12, Col 36]
```

The diagnostic is correct but it should only be emitted once.

This comes up in practice for `dataclasses.is_dataclass` in typeshed. There is an open issue #3443 about ty incorrectly handling calls to that function. My Codex thinks that is related to the duplicate diagnostics here but I haven't fully confirmed that that's the case. Typeshed's code for `is_dataclass` is definitely dubious.

### Version

0.0.58

**Comment by carljm** [comment]
https://github.com/astral-sh/ruff/pull/26566 fixes #3443, and I'm not sure how it would be related to the duplicate diagnostics? But maybe it is, somehow.
